tst/com/amazon/corretto/crypto/provider/test/RsaCipherTest.java

```diff
@@ -518,13 +588,54 @@ public void native2jceOaepParams(final Integer keySize) throws GeneralSecurityEx
         nativeC.init(Cipher.ENCRYPT_MODE, keyPair.getPublic(), params);
         jceC.init(Cipher.DECRYPT_MODE, keyPair.getPrivate(), params);
 
-        final byte[] plaintext = getPlaintext(keySize / 8 - 42);
+        final int paddingSize = getPaddingSize(params.getParameterSpec(OAEPParameterSpec.class));
+        final byte[] plaintext = getPlaintext(keySize / 8 - paddingSize);
 
         final byte[] ciphertext = nativeC.doFinal(plaintext);
         final byte[] decrypted = jceC.doFinal(ciphertext);
         assertArrayEquals(plaintext, decrypted);
     }
 
+    @ParameterizedTest
+    @MethodSource("digestsXdigestsXlengthParams")
+    public void native2jceOaepParamSpecs(final String md1, final String md2, final Integer keySizeBits) throws GeneralSecurityException {
+        final int keySize = keySizeBits / 8;
```

Comment:
NitPick:

```suggestion
        final int keySize = (keySizeBits + 7) / 8;
```

Reply:
ah, whoops. copy pasta. will include in next rev.